### PR TITLE
Support of compileOnly scope in buildInit plugin

### PR DESCRIPTION
**Pull request by glefloch**
Issue: #1533

Any of the checked boxes below indicate that I took action:

- [x] Reviewed the [Contribution Guidelines](https://github.com/gradle/gradle/blob/master/.github/CONTRIBUTING.md).
- [x] Signed the [Gradle CLA](http://gradle.org/contributor-license-agreement/).
- [x] Ensured that basic checks pass: `./gradlew quickCheck`

For all non-trivial changes that modify the behavior or public API:

- [ ] Before submitting a pull request, I started a discussion on the [Gradle developer list](https://groups.google.com/forum/#!forum/gradle-dev),
      the [forum](https://discuss.gradle.org/) or can reference a [JIRA issue](https://issues.gradle.org/secure/Dashboard.jspa).
- [ ] I considered writing a design document. A design document can be
brief but explains the use case or problem you are trying to solve,
touches on the planned implementation approach as well as the test cases
that verify the behavior. Optimally, design documents should be submitted
as a separate pull request. [Samples](https://github.com/gradle/gradle/tree/master/design-docs)
can be found in the Gradle GitHub repository. Please let us know if you need help with
creating the design document. We are happy to help!
- [x] The pull request contains an appropriate level of unit and integration
test coverage to verify the behavior. Before submitting the pull request
I ran a build on my local machine via the command
`./gradlew quickCheck <impacted-subproject>:check`.
- [ ] The pull request updates the Gradle documentation like user guide,
DSL reference and Javadocs where applicable.

### CODE REVIEW

**Comment by otrosien** [comment]
@glefloch let's check if this can also be tested as part of `MavenProjectsCreatorSpec.groovy`

**Comment by glefloch** [comment]
@otrosien is there a better way to convert the `build.gradle` file content into an object instead of just checking if the file contains the right string?

**Comment by otrosien** [comment]
@glefloch the substring check is good enough if you ask me... Otherwise we would probably use the `org.gradle.testfixtures.ProjectBuilder`, pass the build.gradle to it, and then inspect the result.

**Comment by otrosien** [comment]
Changes look good @glefloch. Ready for @gradle devs to review.

**Comment by bmuschko** [comment]
@glefloch Thanks for providing the pull request. Did you sign the CLA? I couldn't find your name in the list of CLA submission.

File: subprojects/build-init/src/main/groovy/org/gradle/buildinit/plugins/internal/maven/Maven2Gradle.groovy [hunk 1]
```diff
@@ -307,18 +306,12 @@ version = '$project.version'""";
...
-                       This behavior is not yet supported by Gradle, so this dependency has been converted to a compile dependency.
-                       Please review and delete this closure when resolved. */
-                       '''.stripIndent(16)
+                    scope = 'compileOnly'
                 }
                 def exclusions = mavenDependency.exclusions.exclusion
-                if (exclusions.size() > 0 || providedMessage != "") {
-                    createComplexDependency(mavenDependency, sb, scope, providedMessage)
+                if (exclusions.size() > 0) {
+                    createComplexDependency(mavenDependency, sb, scope, "")
```
Comment (bmuschko) [inline subprojects/build-init/src/main/groovy/org/gradle/buildinit/plugins/internal/maven/Maven2Gradle.groovy]: We should simply remove the last parameter from the method signature plus the handling of the `providedMessage` within `createComplexDependency`. It's simply not needed anymore.

File: subprojects/build-init/src/test/groovy/org/gradle/buildinit/plugins/internal/maven/MavenProjectsCreatorSpec.groovy [hunk 2]
```diff
@@ -113,4 +113,31 @@ class MavenProjectsCreatorSpec extends Specification {
...
+        <artifactId>build-init</artifactId>
+        <version>1.0.0</version>
+        <scope>provided</scope>
+    </dependency>
+  </dependencies>
+</project>"""
+        def mavenProjects = creator.create(settings.buildSettings(), pom)
+        def converter = new Maven2Gradle(mavenProjects, temp.root.getCanonicalFile())
+
+        when:
```
Comment (bmuschko) [inline subprojects/build-init/src/test/groovy/org/gradle/buildinit/plugins/internal/maven/MavenProjectsCreatorSpec.groovy]: Can you align the statements below `when` and `then` as being use in other test methods? For better readability can you also add a new line between `when` and `then`?

File: subprojects/build-init/src/test/groovy/org/gradle/buildinit/plugins/internal/maven/MavenProjectsCreatorSpec.groovy [hunk 3]
```diff
@@ -113,4 +113,31 @@ class MavenProjectsCreatorSpec extends Specification {
         def ex = thrown(MavenConversionException)
         ex.message == "Unable to create Maven project model. The POM file $pom does not exist."
     }
+
+    def "creates projects with compileOnly library"() {
```
Comment (bmuschko) [inline subprojects/build-init/src/test/groovy/org/gradle/buildinit/plugins/internal/maven/MavenProjectsCreatorSpec.groovy]: Should probably be renamed to something more expressive like "can translate dependency assigned to Maven provided scope into compileOnly".

**Comment by glefloch** [comment]
Hi, Sorry I thought I signed it. I just did it. And I also push code review fixes.

**Comment by bmuschko** [comment]
@glefloch Thanks for changes! It has been merged. @otrosien Thanks for the review.

**Comment by lacasseio** [comment]
Everything looks good to me!

**Pull request merged**